packages/babel-generator/src/generators/jsx.js

```diff
@@ -1,8 +1,9 @@
 export function JSXAttribute(node: Object) {
   this.print(node.name, node);
   if (node.value) {
+    const raw = node.extra && node.extra.raw;
     this.token("=");
-    this.print(node.value, node);
+    this.print(raw || node.value, node);
```

Comment:
Hmm, I'd intentionally avoided that because of the  value and rawValue comparison, thinking that they'd be unequal in exactly the scenarios this PR attempts to address. But I didn't actually test that, and of course your point is valid, so I'll try to find a way to address it (may need to compare an escaped version of value or similar). 

Thanks for the helpful catch @loganfsmyth !